Square Autos: Pathing + Drive Relative

TimB-87 opened the pull request:
Created the drive-in-a-square auto using two methods: 
1. The drive relative function (we just need to plug in x and y values for speed but I don't entirely understand how it works) and 
2. Pathing (after creating new field points specifically for this) - though pathing didn't work last year so I'm not sure if it would work for this but I like the idea of trying to figure it out

TimB-87 committed:
Square Autos: Pathing + Drive Relative

bot190 commented on src/main/java/frc/robot/auto/AutoPath.java:
```diff
@@ -140,6 +140,15 @@ public enum AutoPath {
             new Pose2d()),
 
+    // Square Auto Pathing
+    SQUARE_AUTO_POS1(0.2, new Pose2d(1.0, 1.0, AutoConstants.ROTATION_0)),
```
Unless we have something to set the robots initial position I think this will make the robot move 1m in x and y before starting to drive the box..might want to make this 0,0?

TimB-87 replied:
Hm, I'll make it 0,0 then. I didn't quite make the connection that it should start at zero.

msoucy commented on src/main/java/frc/robot/Auto.java:
```diff
@@ -216,7 +207,8 @@ public CommandBase combinedAuto(ConeStation conePos, CubePickupLocation cubePos,
                 scoreCube()
-        // position to be in front of drive station in community? I'll add that
+        // position to be in front of drive station in community? I'll add that <- ha
+        // you thought, no becuase this auto is useless
```
If we're going to be snarky, can we be snarky with the correct spelling of "because"?

bot190 commented on src/main/java/frc/robot/Auto.java:
```diff
@@ -227,4 +219,33 @@ private CommandBase timingWait() {
+    public CommandBase squareAutoDriveRelative() {
+        return sequence(
+                drive.setGyro180(),
+                drive.driveRelative(new Translation2d(0, 0), 0, 3),
```
I believe the `Translation2d` here is the coordinates to move to relative the current position. The second argument is the angle the robot should be facing when it gets there. So these commands will just cause the robot to spin around.

TimB-87 replied:
Ah, alright, I will update this with coordinates there.

TimB-87 committed:
Update square autos with correct values

Translation2d takes coordinates, Pose2d takes distance on axes, I flipped the both of them in the last iteration so it is fixed with this one.

TimB-87 committed:
Adjust and Create Autos

Adjust Square auto coordinates, create Triangle auto and Knockout auto. Also begin attempting a method to reset the coordinates.

TimB-87 committed:
Small auto updates

TimB-87 committed:
Intake command updates

Hsaunders603 committed:
some wheel test changes -- shouldn't alter any existing code

Liammurray19 committed:
Made positions to drive in a rectangle (and rotate)

TimB-87 committed:
Update Pickup Auto (ex-knockout auto)

TimB-87 committed:
Merge remote-tracking branch 'origin/new-auto' into new-auto

TimB-87 committed:
Fix merge conflicts fingers crossed)

TimB-87 committed:
Auto Updates

TimB-87 committed:
Revert "Auto Updates"

This reverts commit 3c52c3f72f71941b21adc5d98422f63cd9ae26d7.

TimB-87 committed:
Some makeshift auto updates and some PathPlanner waypoints

msoucy commented:
Merge conflicts, but we don't really care about it at this point since we have a 2024 robot
